src/scripts/main.js

```diff
@@ -1,3 +1,41 @@
 'use strict';
 
 // write your code here
+function success() {
+  const message = document.createElement('div');
+  message.textContent = 'Promise was resolved!';
+  document.querySelector('body').appendChild(message);
+}
+
+function error() {
+  const message = document.createElement('div');
+  message.textContent = 'Promise was rejected!';
+  document.querySelector('body').appendChild(message);
+}
+
+const promise1 = new Promise((resolve, reject) => {
+  const button = document.querySelector('h1');
+
+  button.addEventListener('click', () => {
+    if (button) {
```

Comment:
The condition `if (button)` will always be true if the button exists. This logic is flawed because it doesn't actually check if the button was clicked. Consider revising this logic to ensure the promise resolves only when the button is clicked.